jcl/src/java.base/share/classes/java/lang/StringBuffer.java

```diff
@@ -529,8 +529,59 @@ public StringBuffer append (float value) {
  * @param		value	the integer
  * @return		this StringBuffer
  */
-public StringBuffer append (int value) {
-	return append(Integer.toString(value));
+public synchronized StringBuffer append(int value) {
+	/*[IF Sidecar19-SE]*/
+	/*[IF Sidecar19-SE-B174]*/
+	int currentLength = lengthInternalUnsynchronized();
+	int currentCapacity = capacityInternal();
+			
+	int valueLength = Integer.stringSize(value);
+			
+	int newLength = currentLength + valueLength;
+			
+	if (newLength > currentCapacity) {
+		ensureCapacityImpl(newLength);
+	}
+
+	if (String.enableCompression && count >= 0) {
+		Integer.getChars(value, newLength, this.value);
+	} else {
+		StringUTF16.getChars(value, newLength, this.value);
+	}
+
+	count = newLength;
+			
+	return this;
+	/*[ELSE]*/
+	return append (String.valueOf (value));
+	/*[ENDIF]*/
+	/*[ELSE]*/
+	if (value != Integer.MIN_VALUE) {
+		if (String.enableCompression && count >= 0) {
+			return append (String.valueOf (value));
+		} else {
+			int currentLength = lengthInternalUnsynchronized();
+			int currentCapacity = capacityInternal();
+			
+			int valueLength = value < 0 ? Integer.stringSize(-value) + 1 : Integer.stringSize(value);
```

Comment:
Why did you not simplify this use of Integer.stringSize() like the others?

Reply:
The Java 8 JCL implementation of `Integer.stringSize` does not handle negative integers. This was only fixed in Java 9 JCL.